packages/flutter/lib/src/widgets/window.dart

```diff
@@ -2,5 +2,307 @@
+  /// set previously.
+  ///
+  /// System compositor is free to ignore the request.
+  void setContentSize(WindowSizing contentSize);
+
+  /// Request change for the window title.
+  /// [title] new title of the window.
+  void setTitle(String title);
+
+  /// Request change for the window state.
+  /// [state] new state of the window.
+  void setState(WindowState state);
+}
+
+/// [WindowingOwner] is responsible for creating and managing window controllers.
+///
+/// Custom subclass can be provided by subclassing [WidgetsBinding] and
+/// and overriding the [createWindowingOwner] method.
+abstract class WindowingOwner {
+  /// Creates a [RegularWindowController] with the provided properties.
+  RegularWindowController createRegularWindowController({
+    required WindowSizing contentSize,
+    required RegularWindowControllerDelegate delegate,
+  });
+
+  /// Returns whether application has any top level windows created by this
+  /// windowing owner.
+  bool hasTopLevelWindows();
+
+  /// Creates default windowing owner for standard desktop embedders.
+  static WindowingOwner createDefaultOwner() {
+    return window_impl.createDefaultOwner() ?? _FallbackWindowingOwner();
+  }
+}
+
+/// Windowing delegate used on platforms that do not support windowing.
+class _FallbackWindowingOwner extends WindowingOwner {
+  @override
+  RegularWindowController createRegularWindowController({
+    required WindowSizing contentSize,
+    required RegularWindowControllerDelegate delegate,
+  }) {
+    throw UnsupportedError(
+      'Current platform does not support windowing.\n'
+      'Implement a WindowingDelegate for this platform.',
+    );
+  }
+
+  @override
+  bool hasTopLevelWindows() {
+    return false;
+  }
+}
+
+/// The [RegularWindow] widget provides a way to render a regular window in the
+/// widget tree. The provided [controller] creates the native window that backs
+/// the widget. The [child] widget is rendered into this newly created window.
+///
+/// While the window is being created, the [RegularWindow] widget will render
+/// an empty [ViewCollection] widget. Once the window is created, the [child]
+/// widget will be rendered into the window inside of a [View].
+///
+/// An example usage might look like:
+/// ```dart
+/// final RegularWindowController controller = RegularWindowController(
+///   size: const Size(800, 600),
+///   sizeConstraints: const BoxConstraints(minWidth: 640, minHeight: 480),
+///   title: "Example Window",
+/// );
+/// runApp(RegularWindow(
+///   controller: controller,
+///   child: MaterialApp(home: Container())));
+/// ```
+///
+/// When a [RegularWindow] widget is removed from the tree, the window that was created
+/// by the [controller] is automatically destroyed if it has not yet been destroyed.
+///
+/// Widgets in the same tree as the [child] widget will have access to the
+/// [RegularWindowController] via the [WindowControllerContext] widget.
```

Comment:
Should we refer to `[RegularWindowController]` for details, rather than repeating the same?

Or the other way 'round?

Reply:
I would leave it as it is. Maybe it's better to have some duplication rather than having user jump around. No strong feelings about it though.